fineract-provider/src/main/java/org/apache/fineract/cob/loan/SetLoanDelinquencyTagsBusinessStep.java

```diff
@@ -74,6 +96,21 @@ public Loan execute(Loan loan) {
         return loan;
     }
 
+    private boolean isDelinquencyOnPause(Loan loan, List<LoanDelinquencyActionData> effectiveDelinquencyList) {
+        LocalDate cobBusinessDate = ThreadLocalContextUtil.getBusinessDateByType(BusinessDateType.COB_DATE);
+        boolean isPaused = isPausedOnDate(cobBusinessDate, effectiveDelinquencyList);
+        boolean wasPausedOneDayBefore = isPausedOnDate(cobBusinessDate.minusDays(1), effectiveDelinquencyList);
+        if ((isPaused && !wasPausedOneDayBefore) || (!isPaused && wasPausedOneDayBefore)) {
+            businessEventNotifierService.notifyPostBusinessEvent(new LoanDelinquencyRangeChangeBusinessEvent(loan));
+        }
```

Comment:
Frankly speaking this is causing a side-effect in the method while its name is indicating it's just a simple check. Don't you think this is a bit misleading?